hammerblade/torch/kernel/hb_parallel_for.hpp

```diff
@@ -22,9 +22,8 @@ inline uint32_t offset_calc(uint32_t idx, HBTensor<scalar_t> tensor) {
   uint32_t factor = 1;
   uint32_t offset = 0;
   for(int32_t i = tensor.ndim() - 1; i >= 0; i--) {
-    uint32_t dimx = idx % sizes[i];
+    uint32_t dimx = (idx / factor) % sizes[i];
     offset += dimx * strides[i];
-    idx /= factor;
     factor *= sizes[i];
```

Comment:
Cool! In many programs though, I've seen semantics like below for index calculation (`%` paired with `/`), so might be easily recognized and understood:
```
    uint32_t dimx = idx % sizes[i];
    idx /= sizes[i];
    offset += dimx * strides[i];
```

Reply:
Actually we save multiplies with above code? But probably, compiler is smart enough is remove the extra multiply...

Reply:
ok! one fewer integer multiplication! updated the code.
just make sure I'm doing it correctly ;)

Reply:
Really?! Compiler didn't optimize it out?

Reply:
Looks good to me btw!

Reply:
oh I didn't check assembly ;)
yeah I assume compilers are smart

Reply:
Agree, if we can fix it in source, why bother...